ruler: Add flag to configure the queue for pending Alertmanager notif…

roth-wine opened the pull request:
<!--
    Keep PR title verbose enough and add prefix telling
    about what components it touches e.g "query:" or ".*:"
-->

<!--
    Don't forget about CHANGELOG!

    Changelog entry format:
    - [#<PR-id>](<PR-URL>) Thanos <Component> ...

    <PR-id> Id of your pull request.
    <PR-URL> URL of your PR such as https://github.com/thanos-io/thanos/pull/<PR-id>
    <Component> Component affected by your changes such as Query, Store, Receive.
-->

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Add flag to configure the queue for pending Alertmanager notifications
In the [prometheus package](https://github.com/prometheus/prometheus/blob/7789ef27c838a567edd65b5fb767d4a35cab91ac/cmd/prometheus/main.go#L532) we are already able to configure the capacity size. The fixed value could lead to issues sending alerts when evaluating a huge amount of alerts in one ruler instance.

roth-wine commented:
Failing test seems unrelated to my changes and might be flacky.

GiedriusS reviewed (commented):
How would the user or anyone know what should be a "good" value? Could we provide some guidance in the README?

roth-wine commented:
I have made the flag hidden :smile:

roth-wine commented:
Failing test seems again unrelated to my changes and might be flacky.

roth-wine committed:
ruler: Add flag to configure the queue for pending Alertmanager notifications

In the [prometheus package](https://github.com/prometheus/prometheus/blob/7789ef
27c838a567edd65b5fb767d4a35cab91ac/cmd/prometheus/main.go#L532) we are already a
ble to configure the capacity size. The fixed value could lead to issues
sending alerts when evaluating a huge amount of alerts in one ruler
instance.

Signed-off-by: roth-wine <git@roth-wine.de>

roth-wine commented:
@GiedriusS I have rebased this onto main, resolved the conflicts and the CI is green.

Is this something we can get merged, or are there any remaining issues on it?
The only review comment (make the flag hidden) has been addressed since May 2025.
I am happy to close it if it's not a direction Thanos wants.
